test-workflows/tasks/generators/clusterloader/load.yaml

```diff
@@ -0,0 +1,127 @@
+  - name: cluster-name
+    description: name of the cluster 
+    default: hakuna-matata
+  - name: region
+    description: region of the cluster
+    default: us-west-2
+  - name: endpoint
+    description: "aws eks enpoint"
+    default: https://api.us-west-2.wesley.amazonaws.com 
+  - name: nodes-per-namespace
+    description: "nodes per namespace to get created for load test "
+    default: 10
+  - name: cl2-load-test-throughput
+    description: " throughput used for mutate operations"
+    default: 10
+  - name: pods-per-node
+    description: "pod density"
+    default: 10
+  - name: nodes
+    description: "number of dataplane nodes to run the load test against"
+    default: 100
+  workspaces:
+  - name: source
+  - name: config
+  - name: secrets
+    mountPath: /tekton/home/.aws
+  steps:
+  - name: write-kubeconfig
+    image: amazon/aws-cli
+    workingDir: $(workspaces.config.path)
+    script: |
+      aws sts get-caller-identity
+      # get the kube config if not present in default location using 
+      if [ ! -e $(workspaces.config.path)/kubeconfig ];then
+          echo "kube config"
+          aws eks --endpoint $(params.endpoint) update-kubeconfig --name $(params.cluster-name) --region $(params.region)
+          cp -R /root/.kube/config  $(workspaces.config.path)/kubeconfig
+      fi    
+  - name: prepare-loadtest      
+    image: docker.io/library/golang:$(params.version)
+    workingDir: $(workspaces.source.path)
+    script: |
+      git clone $(params.giturl)
+      if [ ! -e $GOPATH/src/k8s.io/perf-tests/clusterloader2/go.mod ];then
+         SRC_PATH="$GOPATH/src/k8s.io/perf-tests"
+         mkdir -p $SRC_PATH
+         cp -R $(workspaces.source.path)/perf-tests/* $SRC_PATH
+         cd $SRC_PATH
+      fi
+      cd clusterloader2
+      GOPROXY=direct go build -o $(workspaces.source.path)/tmp/clusterloader  ./cmd/
+  - name: run-loadtest
+    image: amazon/aws-cli
+    workingDir: $(workspaces.source.path)
+    script: |
+      cat > "$(workspaces.source.path)/tmp/overrides.yaml" <<EOL
+      NODES_PER_NAMESPACE: $(params.nodes-per-namespace)
+      CL2_LOAD_TEST_THROUGHPUT: $(params.cl2-load-test-throughput)
+      CL2_USE_HOST_NETWORK_PODS: false
+      PODS_PER_NODE: $(params.pods-per-node)
+      SMALL_STATEFUL_SETS_PER_NAMESPACE: 0
+      MEDIUM_STATEFUL_SETS_PER_NAMESPACE: 0
+      CL2_ENABLE_PVS: false
+      PROMETHEUS_SCRAPE_KUBE_PROXY: false
+      ENABLE_SYSTEM_POD_METRICS: false
+      NODE_MODE: master 
+      EOL
+      RESULT_DIR=$(workspaces.source.path)/tmp/results
+      mkdir -p ${RESULT_DIR} 
+      mkdir -p /root/.kube/
+      cp $(workspaces.config.path)/kubeconfig /root/.kube/config
+      cp -R /tekton/home/.aws /root/.aws
+      export PATH=/tekton/home/.aws:$PATH
+      curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+      install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+      kubectl version
```

Comment:
Why all these kubectl commands?

Reply:
It's good to have logs about basic details of the cluster before kicking off a test. Helpful especially during debug.

Reply:
I'd add a comment so readers know it isn't necessary for some reason (which can get confusing)

Reply:
This should be easily observable via the tekton task itself. I want to avoid unnecessary complexity and noise.